src/Npgsql/Replication/TestDecoding/TestDecodingAsyncEnumerable.cs

```diff
@@ -46,12 +47,36 @@ async IAsyncEnumerator<TestDecodingData> StartReplicationInternal(CancellationTo
             _slot, cancellationToken, _walLocation, _options.GetOptionPairs());
         var encoding = _connection.Encoding!;
 
-        await foreach (var msg in stream.WithCancellation(cancellationToken))
+        var buffer = ArrayPool<byte>.Shared.Rent(4096);
+
+        try
+        {
+            await foreach (var msg in stream.WithCancellation(cancellationToken))
+            {
+                var len = (int)msg.Data.Length;
```

Comment:
```suggestion
                var len = (int)msg.Data.Length;
                Debug.Assert(msg.Data.Position == 0);
```

Or just always decrement it.